internal/ingress/controller/controller_test.go

```diff
@@ -586,6 +598,338 @@ func TestExtractTLSSecretName(t *testing.T) {
 	}
 }
 
+func TestGetBackendServers(t *testing.T) {
```

Comment:
`getBackendServers` does a lot of things unfortunately. And this test is a very specific test - it's making sure the servers are generated correctly when there's canary ingress.

But without refactoring most of `controller.go` I don't have a better idea as to what to call this test function.

Reply:
Yup that's what I was thinking when writing it, still, other tests can be added for other cases in the future here.